workflow/controller/workflowpod.go

```diff
@@ -1150,6 +1158,14 @@ func createSecretVolume(volMap map[string]apiv1.Volume, art wfv1.Artifact, keyMa
 		createSecretVal(volMap, art.OSS.SecretKeySecret, keyMap)
 	} else if art.GCS != nil {
 		createSecretVal(volMap, art.GCS.ServiceAccountKeySecret, keyMap)
+	} else if art.HTTP != nil {
+		createSecretVal(volMap, art.HTTP.BasicAuth.UsernameSecret, keyMap)
+		createSecretVal(volMap, art.HTTP.BasicAuth.PasswordSecret, keyMap)
+		createSecretVal(volMap, art.HTTP.ClientCert.ClientCertSecret, keyMap)
+		createSecretVal(volMap, art.HTTP.ClientCert.ClientKeySecret, keyMap)
+		createSecretVal(volMap, art.HTTP.OAuth2.ClientIDSecret, keyMap)
+		createSecretVal(volMap, art.HTTP.OAuth2.ClientSecretSecret, keyMap)
+		createSecretVal(volMap, art.HTTP.OAuth2.TokenURLSecret, keyMap)
```

Comment:
these lines (and the ones in the `createArchiveLocationSecret`) were not added in the recent update of the http artifact. But for my understanding they are needed, so i addded them. Correct me if im wrong.